src/analysis/special_functions/trigonometric/angle.lean

```diff
@@ -271,6 +272,131 @@ cos_antiperiodic θ
+begin
+  rw [to_real_coe, to_Ioc_mod_eq_self two_pi_pos],
+  ring_nf
+end
+
+lemma to_real_coe_eq_self_iff_mem_Ioc {θ : ℝ} : (θ : angle).to_real = θ ↔ θ ∈ set.Ioc (-π) π :=
+by rw [to_real_coe_eq_self_iff, ←set.mem_Ioc]
+
+lemma to_real_injective : function.injective to_real :=
+begin
+  intros θ ψ h,
+  induction θ using real.angle.induction_on,
+  induction ψ using real.angle.induction_on,
+  simpa [to_real_coe, to_Ioc_mod_eq_to_Ioc_mod, zsmul_eq_mul, mul_comm _ (2 * π),
+         ←angle_eq_iff_two_pi_dvd_sub, eq_comm] using h,
+end
+
+@[simp] lemma coe_to_real (θ : angle): (θ.to_real : angle) = θ :=
+begin
+  induction θ using real.angle.induction_on,
+  exact coe_to_Ioc_mod _ _
+end
+
+lemma neg_pi_lt_to_real (θ : angle) : -π < θ.to_real :=
+begin
+  induction θ using real.angle.induction_on,
+  exact left_lt_to_Ioc_mod _ two_pi_pos _
+end
+
+lemma to_real_le_pi (θ : angle) : θ.to_real ≤ π :=
+begin
+  induction θ using real.angle.induction_on,
+  convert to_Ioc_mod_le_right _ two_pi_pos _,
+  ring
+end
+
+lemma abs_to_real_le_pi (θ : angle) : |θ.to_real| ≤ π :=
+abs_le.2 ⟨(neg_pi_lt_to_real _).le, to_real_le_pi _⟩
+
+lemma to_real_mem_Ioc (θ : angle) : θ.to_real ∈ set.Ioc (-π) π :=
+begin
+  induction θ using real.angle.induction_on,
+  convert to_Ioc_mod_mem_Ioc _ two_pi_pos _,
+  ring
+end
+
+@[simp] lemma to_Ioc_mod_to_real (θ : angle): to_Ioc_mod (-π) two_pi_pos θ.to_real = θ.to_real :=
+begin
+  induction θ using real.angle.induction_on,
+  rw to_real_coe,
+  exact to_Ioc_mod_to_Ioc_mod _ _ _ _
+end
+
+@[simp] lemma to_real_zero : (0 : angle).to_real = 0 :=
+begin
+  rw [←coe_zero, to_real_coe_eq_self_iff],
+  exact ⟨(left.neg_neg_iff.2 real.pi_pos), real.pi_pos.le⟩
+end
+
+@[simp] lemma to_real_eq_zero_iff {θ : angle} : θ.to_real = 0 ↔ θ = 0 :=
+begin
+  nth_rewrite 0 ←to_real_zero,
+  exact to_real_injective.eq_iff
+end
+
+@[simp] lemma to_real_pi : (π : angle).to_real = π :=
+begin
+  rw [to_real_coe_eq_self_iff],
+  exact ⟨left.neg_lt_self real.pi_pos, le_refl _⟩
+end
+
+@[simp] lemma to_real_eq_pi_iff {θ : angle} : θ.to_real = π ↔ θ = π :=
+begin
+  nth_rewrite 0 ←to_real_pi,
+  exact to_real_injective.eq_iff
+end
+
+lemma abs_to_real_coe_eq_self_iff {θ : ℝ} : |(θ : angle).to_real| = θ ↔ 0 ≤ θ ∧ θ ≤ π :=
+⟨λ h, h ▸ ⟨abs_nonneg _, abs_to_real_le_pi _⟩, λ h,
+ (to_real_coe_eq_self_iff.2 ⟨(left.neg_neg_iff.2 real.pi_pos).trans_le h.1, h.2⟩).symm ▸
+    abs_eq_self.2 h.1⟩
+
+lemma abs_to_real_neg_coe_eq_self_iff {θ : ℝ} : |(-θ : angle).to_real| = θ ↔ 0 ≤ θ ∧ θ ≤ π :=
+begin
+  refine ⟨λ h, h ▸ ⟨abs_nonneg _, abs_to_real_le_pi _⟩, λ h, _⟩,
+  by_cases hnegpi : θ = π, { simp [hnegpi, real.pi_pos.le] },
+  rw [←coe_neg, to_real_coe_eq_self_iff.2 ⟨neg_lt_neg (lt_of_le_of_ne h.2 hnegpi),
+                                           (neg_nonpos.2 h.1).trans real.pi_pos.le⟩, abs_neg,
+      abs_eq_self.2 h.1]
+end
+
+@[simp] lemma sin_to_real (θ : angle) : real.sin θ.to_real = sin θ :=
```

Comment:
This and the next lemma should be `norm_cast` too, I believe.

Reply:
No, because `norm_cast` only deals with coercions, not with other coercion-like functions such as `to_real`, so marking it `norm_cast` produces an error `norm_cast: badly shaped lemma, lhs must contain at least one coe`.